### ISSUE TITLE
VSCode  is "Unable to open 'Extension: ': An unknown error occurred."

### ISSUE THREAD

**Issue by theSeifHub** [microsoft/vscode#95427]
- OS Version: Windows10 x64

Steps to Reproduce:
1. Install VSCode & Open for the 1st time
2. Search for an extension in the market place and click on it

Does this issue occur when all extensions are disabled?: Yes

### - VSCode Version: 1.44 (March 2020) & Insiders 1.45
Whenever I search for an extension and try to open to install it,  it let me peek quickly at the extension page before closing it and displaying this notification:

> Unable to open 'Extension: _EXTENSION NAME_': An unknown error occurred. Please consult the log for more details.
![VSCODE ](https://user-images.githubusercontent.com/12737649/79440382-36f55c80-7fd6-11ea-996f-968d6b6a6029.jpg)

### - VSCode Version: 1.42 (January 2020)

Now, when I uninstalled v1.44 and reinstalled v1.42, the extension page opens but when trying to install, it displays the following:
> Failed to install '_CREATOR.EXTENSION_'. Download Manually.
![vscode 1 42](https://user-images.githubusercontent.com/12737649/79442170-7a50ca80-7fd8-11ea-8b36-1f13a947d056.jpg)

Downloading manually and installing works fine with this version, but still do not understand why this is happening.

**Searched for a solution but found nothing applicable. Why is this happening and how to fix it?**

**Comment by gjsjohnmurray** [comment]
Did you look at #94858 in case it applies to you?

**Comment by theSeifHub** [comment]
Yes, I did before posting the issue. and it doesn't apply, as I'm not using AV software nor having any issue with Extension host log.

**Comment by chrmarti** [comment]
@theSeifHub Are you behind a proxy? What's in the Network tab of the dev tools? (`Help` > `Toggle Developer Tools`)

**Comment by theSeifHub** [comment]
How do I know if I'm behind a proxy?
I found some answers on StackOverflow that suggest it has something to do with proxy, but I couldn't relate, so I didn't follow along.
![network](https://user-images.githubusercontent.com/12737649/79547491-46d37600-8094-11ea-9145-a7f95c300c17.jpg)

**Comment by chrmarti** [comment]
Proxies are sometimes used in corporate setups. Search for 'proxy' in Chrome's settings and follow its link to check if you have a proxy configured in the OS.

**Comment by Zorono** [comment]
those problems happens with me too...

VS Code version: Code 1.44.2 (ff915844119ce9485abfe8aa9076ec76b5300ddd, 2020-04-16T16:33:57.013Z)
OS version: Windows_NT ia32 6.3.9600

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz (4 x 2594)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: unavailable_off<br>protected_video_decode: unavailable_off<br>rasterization: unavailable_off<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: unavailable_off|
|Load (avg)|undefined|
|Memory (System)|2.63GB (0.40GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
code-gnu-global|aus|0.2.2
vscode-intelephense-client|bme|1.3.11
php-debug|fel|1.13.0
node-module-intellisense|lei|1.5.0
cmake-tools|ms-|1.3.1
cpptools|ms-|0.27.0
material-icon-theme|PKi|4.0.1
vscode-pawn|sou|0.6.6
vscode-arduino|vsc|0.3.0


</details>

**Comment by Zorono** [comment]
My `Dev Tools -> Network`
![image](https://user-images.githubusercontent.com/19735243/79927720-eac87300-8440-11ea-958e-498c8ede8482.png)

My ISP's DNS servers
```
163.121.128.134
163.121.128.135
```

**Comment by chrmarti** [comment]
Could you start VS Code with `--log-net-log=netlog.json`, try to open an extension and then post the created netlog.json here?

**Comment by Zorono** [comment]
[netlog.json(sorry Github didn't allow me to upload JSON files, so i compressed it...)](https://github.com/microsoft/vscode/files/4519472/netlog.json.zip)

and my `Code.exe --log-net-log=netlog.json` command's output in the Command Prompt...
```

 Warning: 'log-net-log' is not in the list of known options, but still passed to
 Electron/Chromium.
[main 2020-04-22T23:37:09.673Z] update#setState idle
(node:4948) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:4948) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
[main 2020-04-22T23:37:39.855Z] update#setState checking for updates
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
(node:8172) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
[main 2020-04-22T23:37:44.522Z] update#setState idle
[main 2020-04-22T23:41:12.042Z] update#setState checking for updates
[main 2020-04-22T23:41:19.474Z] update#setState idle
[8176:0423/014127.904:ERROR:CONSOLE(24)] "Empty response arrived for script 'devtools://devtools/remote/serve_file/@c1109b707b1eda5d98eb2ed6bd74083bc352f482/product_registry_impl/product_registry_impl_module.js'", source: devtools://devtools/bundled/shell.js (24)
[8176:0423/014130.097:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014130.097:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014130.098:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014130.099:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014130.099:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014130.099:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.765:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.765:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.805:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.805:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.806:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.806:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.918:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)
[8176:0423/014131.944:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Registry", source: devtools://devtools/bundled/shell.js (109)

```

**Comment by chrmarti** [comment]
@Zorono That looks like everything worked. What doesn't work? Did you do that while recording the net log?

**Comment by deepak1556** [comment]
Proxy wise, the system has proxy auto-detect enabled. Tried to fetch any pac files over DHCP and wpad but no file was available and resorted to direct connections.

But I do see a lot of asses requests terminated with `ERR_NAME_NOT_RESOLVED` , whatever DNS server is being used is unable to resolve these domain names, maybe want to try another public dns server.

But before messing with DNS servers, lets try loading one of the failed requests in chrome browser, @Zorono can you load https://msjsdiag.gallerycdn.vsassets.io/extensions/msjsdiag/debugger-for-chrome/4.12.6/1579885786521/Microsoft.VisualStudio.Services.Icons.Default in your chrome browser and share the output screenshot ?

**Comment by Zorono** [comment]
![image](https://user-images.githubusercontent.com/19735243/80242200-42dfbf00-8665-11ea-98fe-7dc5c7931d4e.png)

**Comment by deepak1556** [comment]
@Zorono just to confirm, did changing the DNS server help ?

**Comment by Zorono** [comment]
i amn't sure exactly whether clearing `DNS Cache` or changing `DNS Servers` is the reason why it works now...
but yes, it is(i was using my ISP's DNS Servers which is s*cking horrible)

**Comment by deepak1556** [comment]
Thanks! @theSeifHub can you try similar solution, since the issue you are facing seems to be same.

**Comment by theSeifHub** [comment]
@deepak1556  my networking vocab and knowledge isn't that great. So just to make sure I understand correctly, I'm supposed to clear DNS cache (of my browser?!), then reload this link you posted above -which gave me a similar "This site can’t be reached" screenshot BTW-??

**Comment by theSeifHub** [comment]
changed DNS server and cleared cache but none worked

**Comment by Zorono** [comment]
@theSeifHub 
كلم دعم شركة الاتصالات الخاصة بيك(WE/Orange/....) لأن فيه مشكلة فال DNS Servers المصرية
علي حسب ما اعتقد

**Comment by chrmarti** [comment]
@theSeifHub Could you start VS Code with `--log-net-log=netlog.json`, try to open an extension and then post the created netlog.json here?

**Comment by theSeifHub** [comment]
@chrmarti 
[netlog.zip](https://github.com/microsoft/vscode/files/4539484/netlog.zip)

**Comment by theSeifHub** [comment]
@Zorono
 طب أقولهم ايه؟ غيرولي سرفرات DNS ؟

**Comment by deepak1556** [comment]
Thanks for the log @theSeifHub , the log contains the same error as https://github.com/microsoft/vscode/issues/95427#issuecomment-618890116

There is nothing VSCode can do here.

@Zorono can you provide the steps for solutions you tried, would help @theSeifHub . Thanks!

**Comment by Zorono** [comment]
@deepak1556 that problem is mostly caused by our **Egyptian Domain Name Servers** which are blocking many of **Microsoft's Servers**(at least that's my explanation of my DNS problem and please forgive my small knowledge for any stupidness)... however

1. I Changed my **Domain Name Servers** from my Router Modem's cPanel for both the `DHCP` and `DNS` to **Google DNS**(Primary: `8.8.8.8` Secondary: `8.8.4.4`)
2. I flushed my Adapter's `DNS Resolver Cache` using the command `ipconfig /flushdns` (Windows)
3. I stopped the `DNS Caching Services` using the command `net stop dnscache` (Windows)
4. Make sure to Change your Adapter's `IPv4 Domain Name Servers`

@theSeifHub حاول تخليهم يظبطوا ال `DNS` بتاعم علشان يشغل ال سيرفرات بتاعت مايكروسوفت
لأنه علي حسب ما اعتقد انهم  عاملين حظر ليها او الله اعلم بس هما مش بيوضوحوا اي حاجة ,كل الي عندهم `يا فندم حضرتك تقدر تغير ل سيرفر تاني زي جوجل لو عندك مشكلة`

**Comment by rioj7** [comment]
The error message says `Please consult the log for more details`.

Where is this log?

Looking in the VSC program files I find this sentence a lot. Does it mean it all goes in one log file or are they using different logs?

The OUTPUT tab has a couple of `Log (xxxx)` entries but none of them shows the needed log error.
Where are the files with these log messages or are they in memory only?

For me a restart of VSC (1.44.2) fixed the problem. I can now see the extension page but I don't see any of the images from the README.md file. The extension icon is displayed.

**Comment by theSeifHub** [comment]
@deepak1556 @chrmarti
I changed the DNS again to Google's a few days ago but forgot to try any extensions then.
Today I opened VSCode, clicked on an extension, viewed its page and images, installed it then uninstalled it to reinstall again. Somehow everything works fine!
Thanks @Zorono you were a great help.

**Comment by ahmedElsehrawy** [comment]
if you didn't fix that problem yet i had that problem today and i searched and got nothing but i discovered that you can add an extension manually by searching the name of extension in vs code and then click install without open the extension details and vs code will show you alert to Download manually click it and it will open the browser and download it manually
![2020-05-08](https://user-images.githubusercontent.com/65005663/81375009-5fdcaf00-90b5-11ea-9c49-780b35d68b12.png)
then click install from VSIX And congrats you have your extension installed

**Comment by theSeifHub** [comment]
@ahmedElsehrawy while vscode 1.42 did, v-1.44 didn't offer any manual download option, I had to visit marketplace.visualstudio on my browser and download each extension, then install from VSIX.
Thanks though, I did already fix the problem by changing my DNS servers

**Comment by philanthr0pist** [comment]
> @deepak1556 that problem is mostly caused by **Egyptian Domain Name Servers** which are blocking many of **Microsoft's Servers**(at least that's my explanation of my DNS problem and please forgive my small knowledge for any stupidness)... however
> 
> 1. I Changed my **Domain Name Servers** from my Router Modem's cPanel for both the `DHCP` and `DNS` to **Google DNS**(Primary: `8.8.8.8` Secondary: `8.8.4.4`)
> 2. I flushed my Adapter's `DNS Resolver Cache` using the command `ipconfig /flushdns` (Windows)
> 3. I stopped the `DNS Caching Services` using the command `net stop dnscache` (Windows)
> 4. Make sure to Change your Adapter's `IPv4 Domain Name Servers`
> 
> in fact, this problem happens with most of VS products...
> i can't update my `Visual Studio 2019` Extensions because of `DNS problems` too...
> 
> @theSeifHub حاول تخليهم يظبطوا ال `DNS` بتاعم علشان يشغل ال سيرفرات بتاعت مايكروسوفت
> لأنه علي حسب ما اعتقد انهم عاملين حظر ليها او الله اعلم بس هما مش بيوضوحوا اي حاجة ,كل الي عندهم `يا فندم حضرتك تقدر تغير ل سيرفر تاني زي جوجل لو عندك مشكلة`

i had the same problem and this answer helped me alot, many thanks.

**Comment by mahmoudelgawad** [comment]
thanks guys , solved by changed to google DNS "8.8.8.8", because of Egypt WE ISP

**Comment by MohamedAnwerEida** [comment]
if you use ubuntu or linux go to [https://stackoverflow.com/questions/61190800/why-is-vscode-unable-to-open-extension-an-unknown-error-occurred](url)
